### PR TITLE
Issue 12 - Game Loop: Part 1 (Dan's Part)

### LINKED ISSUE #12: Game Loop: Part 1
## Feature Overview

- Feature to simulate first player turn and subsequent players turns until turn cycle is complete

### ToDo

-  [x] Functionality to start from turn 1 and prompt player 1 to play cards (DANIEL)
-  [x] Game validating what they played (JUSTIN)
-  [x] Game removing the played cards from their hand, and setting the turns combination. (DANIEL)
-  [x] Moving to player 2, and prompting them to play cards that fit the combination (DANIEL)
-  [x] Validating player 2s choice for valid combo and higher value. (JUSTIN)
-  [ ] Pass removing user from current turns cycle. (JUSTIN)
-  [x] Change numbers array so values 3-15 are not strings, but numbers (JUSTIN)

### PR DESCRIPTION
**Pull request by ziggysauce**
Partially resolves https://github.com/jduong09/thirteen/issues/12
~Branched off of (and blocked by) https://github.com/jduong09/thirteen/pull/11~

## Changes/Additions
- Simulates a working game (only using `single` as the combination type)
- When a player (including the user) submits a combo, the hand will be updated accordingly
- Created the function for setting the combination type
  - NOTE: This isn't truly functional yet because we need more logic for determining what an AI would play. This would only be relevant if the user is turn 1 of a sequences (this logic is still tbd)
- After player X finishes (assuming it's valid), the turn indicator is progressed to the next player
- Right now I'm just simulating a turn for every AI player (see the 3 second timeouts)

## How To Test
- Checkout to `issue-12--game-loop-dan` and pull changes with `git pull`
- Navigate to `localhost:3000` in your browser
- Shuffle the deck. The browser will display a hand of 13 cards as well as indicate which player is going first (whoever has the 3 of spades). Each non-user turn will be simulated until it's the user's turn.

### CODE REVIEW

File: components/gameComponents/hand.js [hunk 1]
```diff
@@ -16,6 +16,13 @@ const Hand = ({ cards, playerTurn, comboIsValid, requestCombo, currentTurnCombo,
   const [hasReset, resetCombo] = useState(false);
   const isMyTurn = playerTurn === 0;
 
+  // Update hand after changes.
+  // NOTE: Apparently this might not be best practice, and some people suggest handling all
+  // state changes from the parent only. But that seems like too much managmement in one place imo.
+  if(cards.length !== hand.length) {
+    setHand(cards);
+  }
```
Comment (jduong09) [inline components/gameComponents/hand.js]: I believe best practice to handle  state change from parent, would be to use a **useEffect** hook to allow the hand component to watch for changes to the **cards** prop, and re-rendering when **cards** changes.

https://react.dev/reference/react/useEffect
--> See Examples of Passing reactive dependencies

`useEffect(() => {
    setHand(cards);
  }, [cards])`

**Comment by jduong09** [comment]
## Review

PR looks good, I'll branch off of it and work on my tasks for #12.

Once #11 is approved by you, I can merge that into main and we can resolve merge conflicts.

File: components/gameComponents/hand.js (hunk 1 above)
Comment (ziggysauce) [reply components/gameComponents/hand.js]: ahh yes, that's probably the right way to handle this

**Comment by jduong09** [comment]
## Update

This branch is not blocked by #11  anymore. This PR looks good, Im going to branch of it and work on my tasks.

**Pull request merged**